### ISSUE TITLE
Expression of type 'dict' cannot be assigned to return type 'Dict'

### ISSUE THREAD

**Issue by TheCleric** [Microsoft/pyright#19]
When using python's type hinting to designate a function as returning a Dict, and returning an actual dict, pyright flags this as invalid, when I believe it is valid (unless I've been doing type hinting wrong, which is always possible).

Minimal reproduction example

```python
from typing import Dict

def return_a_dict() -> Dict:
    return {}
```

**Comment by TheCleric** [comment]
This also goes for List and list.

**Comment by erictraut** [comment]
Thanks for the bug report. I'm working on a fix to add internal aliasing between "Dict" and "dict" and similar analogs.

**Comment by erictraut** [comment]
This is fixed in 1.0.4, which I just published. Thanks for the bug report.
